packages/orchestration/src/orchestration-embedding-response.ts

```diff
@@ -0,0 +1,44 @@
+import type { HttpResponse } from '@sap-cloud-sdk/http-client';
+import type {
+  EmbeddingsPostResponse,
+  EmbeddingResult,
+  EmbeddingsUsage,
+  ModuleResultsBase
+} from './client/api/schema/index.js';
+
+/**
+ * Response wrapper for orchestration embedding requests.
+ */
+export class OrchestrationEmbeddingResponse {
+  public readonly _data: EmbeddingsPostResponse;
+
+  constructor(public readonly response: HttpResponse) {
+    this._data = response.data;
+  }
+
+  /**
+   * Final embedding results.
+   * @returns Array of embedding vectors, where each element is either a number array or a base-64 encoded string of the embedding.
+   */
+  getEmbeddingVectors(): (number[] | string)[] {
```

Comment:
For now, I have allowed both response types, I have opened a discussion with the Orchestration team.

Assumption here is that the service will always only either have a list of float array or a list of encoded strings and not a mix of both, although the return types allow it.